### PR TITLE
Mini-Nukie Update: The Bulldog is a 2-shot burst fire gun edition (and lone ops get some spare ammo for their bulldog)

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

The bulldog shotgun, available to ops, now has a burst fire of 2 rounds. This means it launches a fairly rapid volley of bullets per click.

Currently, the lone operative spawns with a single bulldog shotgun. Instead of just the shotgun, the operative now comes with some spare magazines for their gun.

## Why It's Good For The Game

The bulldog still seems to be really unpopular, and still feels like it isn't quite doing as much as other options for operatives. Even for the cheap price, it just doesn't pull its weight. It lacks staying power or range to face several opponents at once. You can certainly buy ammo to make it better at engaging at range, but this sometimes begs the question as to why you bothered with the bulldog over something else. 

It is primarily only really good at killing one person at a time in close quarters, but the person you would be using that power against is usually a more armored opponent like the Captain or Head of Security. Lots of other nuclear operative weapons simply do that much better than the bulldog.

If the bulldog is meant to be mulching single targets one at a time, it should absolutely be doing that reliably. Because people really like shotguns. So why not unload two buckshot straight into someone's cranium with a single pull of the trigger for a nice crunchy splat.

Lone operative got themselves a bulldog, but were cheated out of the free ammo they would have received if they had simply bought the bulldog from their uplink. Lone ops are not nearly as scary as an actual midround nuclear assault, primarily because their objective doesn't come with the benefit of an infiltrator or a bomb to deliver. They're a punishment mechanic, technically. But even as a punishment, they don't really do a terribly good job at punishing. Since the bulldog is focused on putting down the captain as fast as possible, then it stands to reason that the lone operative gets as much help accomplishing that as necessary.

## Changelog

:cl:
balance: Bulldog Shotguns now have a 2-round burst fire.
balance: Lone Operatives now come with some additional Bulldog Shotgun magazines.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Oh man no that looks like dogshit, I'll figure out how to make it use the shotgun operative outfit for that later

**Comment by jughu** [comment]
bulldog sucks cuz its ass dps compared to other options but you shouldnt buff it instead you should nerf the other options
L6saw is in a perfect spot requires 2 hands 60 brute a shot
Daul wielding c20r is another popular since it beats l6saw damage
Dsword is popular cuz its a energy based station so its super good defensive/offensive option
Where bulldog sucks it makes up for its utility bulldog is the skeleton key of damage types burn/brute/chem or even meteorshot to break into places thats where its power lies which is undervalued

**Comment by DaCoolBoss** [comment]
Needs a single shot mode, mostly to stop accidentally wasting meteor rounds when shooting doors.

**Comment by necromanceranne** [comment]
> bulldog sucks cuz its ass dps compared to other options but you shouldnt buff it instead you should nerf the other options 

Iunno I don't think there is anything wrong with the other guns, to be honest. They're just outclassing the bulldog fairly readily.

**Comment by necromanceranne** [comment]
I've added a firemode selector to the bulldog. I've also placed the shotgun magazines that would be form the weapon case into the lone operative's bag. It doesn't really matter.

Fun little tech I discovered: right clicking in the burst mode swaps the magazine midfire, so you can fire both your primary and alternate ammo. Someone might find that useful.

File: code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 1]
```diff
@@ -188,6 +190,26 @@
 	. += "You can remove a secondary magazine by alt-right-clicking [src]."
 	. += "Right-click to swap the magazine to the secondary position, and vice versa."
 
+/obj/item/gun/ballistic/shotgun/bulldog/ui_action_click(mob/user, actiontype)
+	if(istype(actiontype, /datum/action/item_action/toggle_firemode))
+		burst_select()
+	else
+		..()
+
+/obj/item/gun/ballistic/shotgun/bulldog/proc/burst_select()
```
Comment (Ghommie) [inline code/modules/projectiles/guns/ballistic/shotgun.dm:199]: There's no reason this should be a copypaste of the omonimously named `gun/ballistic/automatic/burst_select()`, or for the original proc to be a children of `gun/ballistic/automatic`, when `burst_size` and `fire_delay` are generic variables that all gun items have.

TL;DR: make a generic `burst_select()` proc for all guns or ballistic guns.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 1 above)
Comment (necromanceranne) [reply code/modules/projectiles/guns/ballistic/shotgun.dm:199]: I got very lazybrain here even though my brain was also telling me 'just make it a generic proc. I'll get it fixed up.
